blob-module/ios/Blob.swift

```diff
@@ -31,15 +31,15 @@ public class Blob: SharedObject {
             return self
         }
         
-        var dataSlice: [String] = []
+        var dataSlice: [BlobPart] = []
         var charsLeft = endIdx - startIdx + 1
         for part in blobParts {
-            if part.count <= charsLeft {
-                charsLeft -= part.count
+            if part.text().count <= charsLeft {
```

Comment:
this doesn't seem like a correct assumption anymore – let's make this work for dataviews, typedarrays ect soon